inference-engine/include/details/ie_mmap_allocator.hpp

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief The header file defines utility PreAllocator class
+ * 
+ * @file ie_pre_allocator.hpp
+ */
+#pragma once
+
+#include <details/ie_exception.hpp>
+#include <memory>
+
+#include "ie_allocator.hpp"
+
+#ifdef _WIN32
+#include <windows.h>
```

Comment:
Please avoid usage of `windows.h` in public headers. It defines dozens of macros with small common names (`min`, `max`, `TEMP`, etc.).